### PR TITLE
Authentication devise

### PR DESCRIPTION
**Pull request by kessio**
Hello 👋🏾 
### Project requirements

- [x] Remove current_user method in ApplicationController, devise will provide us with one.
- [x] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- [x] Can register a new user.
- [x] User logs in with a combination of email and password.
- [x] Hashed password should be stored in the database.
- [x] Ask for confirmation of email.
- [x] Can reset password.
- [x] [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/admin_user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AdminUser < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+end
```
Comment (DammyShittu) [inline app/models/admin_user.rb]: - [x] This devise configuration should be in the User `(models/user.rb)` model. There's no apparent need to create an admin_user model here. Kindly move this config to the User model.

File: db/schema.rb [hunk 2]
```diff
@@ -10,10 +10,26 @@
...
+    t.string "confirmation_token"
+    t.datetime "confirmed_at"
+    t.datetime "confirmation_sent_at"
+    t.string "unconfirmed_email"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["email"], name: "index_admin_users_on_email", unique: true
+    t.index ["reset_password_token"], name: "index_admin_users_on_reset_password_token", unique: true
+  end
+
```
Comment (DammyShittu) [inline db/schema.rb]: - [x] You seem to have generated Devise for a new AdminUser model (which is unnecessary). Kindly note that we are configuring devise for the already existing User model, not AdminUser. Kindly fix that.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,6 +1,13 @@
 # rubocop:disable all
 class ApplicationController < ActionController::Base
-    def current_user
-        User.first
-      end
+    protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```
Comment (DammyShittu) [inline app/controllers/application_controller.rb:6]: - [x] I am able to visit different routes without signing in. Kindly fix that. Currently adding the following helper method fixes that.

  ```
  before_action :authenticate_admin_user!
  ```
  But after fixing the resulting issue with User model, you need to add this helper to this controller to ensure only signed in users can visit certain routes.

  ```
  before_action :authenticate_user!
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @kessio and @shuhad786,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 4]
```diff
@@ -139,12 +139,17 @@ You can deploy this project using:
...
 - LinkedIn: [LinkedIn](https://linkedin.com/in/sharon-kessio-172220b5)
 
+👤 **Shuhad Loofer**
+
+- GitHub: [@Shuhad786](https://github.com/Shuhad786)
+- LinkedIn: [LinkedIn](https://linkedin.com/in/shuhad-loofer)
+
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- FUTURE FEATURES -->
```
Comment (youmari) [inline README.md:155]: - [x] this project should be invalidated by the first reviewer cause the readme file is not professional you shouldn't leave the support and acknowledgments sections empty with placeholders please make sure to add you fill them out with your information
Screenshot 2023-02-09 at 14 33 18

File: app/models/user.rb [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
   has_many :posts, class_name: 'Post', foreign_key: :author_id
   has_many :comments, class_name: 'Comment', foreign_key: :author_id
   has_many :likes, class_name: 'Like', foreign_key: :author_id
 
   validates :name, presence: true
-  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  # validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (youmari) [inline app/models/user.rb]: - [x] it would be better to remove the commented code in order to keep the  file clean and pro

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @kessio , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 
✔️ PRO PR 
✔️ No linters errors
## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @kessio and @shuhad786 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**